docs/DeveloperGuide.md

```diff
@@ -2,37 +2,220 @@
 
 ## Acknowledgements
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+Referred to https://se-education.org/addressbook-level3/DeveloperGuide.html when drafting the DG
 
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### Architecture
 
+![](./images/architeture.png)
+
+The Architecture Diagram given above explains the high-level design of the App.
+
+The **User** sends command which is processed by Input
+
+**Input**: Receives user input string, processes it and sends to CashLeh main app
+
+**CashLeh**: main app
+
+**UI**: Responsible for the printing onto CLI
+
+**Parser**: Process the input string and calls respective methods
+
+**FileStorage**: Reads data from, and writes data to file
+
+### How the architecture components interact with each other
+
+![](./images/main_example.png)
```

Comment:
Also, from the code it seems like the `Command` constructor returns before `execute()` is called? May want to split that up, so that it's not a self-invocation but rather an invocation from the parser.